### ISSUE TITLE
Setting reference folders worsens results for similar images

### ISSUE THREAD

**Issue by anonymous721** [qarmin/czkawka#980]
Setting some folders as Reference Folders leads to worse results than if that checkbox is not ticked. It appears to be symmetric, with it not mattering which folder is set as reference.
For example, I had folders of art from two different sources. Three images were identical to the eye and had the same dimensions, but were PNGs in one folder and JPGs in the other. With neither folder set as reference, all three matches were detected with the similarity threshold set at 6. When either folder was set as reference (regardless of which), then even with the similarity threshold all the way at 40, none of the matches were detected in about 90% of repeated searches (just hitting "search" repeatedly). In about 10% of those searches, one of the matches was detected. but not the other two.

**Comment by qarmin** [comment]
Can you share this files?

Can you test latest binaries from actions - https://github.com/qarmin/czkawka/actions/runs/5225694583 ?

I rewritten comparison part and maybe this will work better than before.

**Comment by anonymous721** [comment]
I no longer have those particular files. 
But the new release definitely works much better. I had a couple folders where the old version was only finding 7 or 8 pairs per search, and different ones each time, but that 6.0.0 release on the same folders found over 100 and seems consistent.

**Comment by anonymous721** [comment]
Turns out there is still odd behaviour. This time I noted down some examples of troublesome files.

Out of the four images in [this post](https://twitter.com/himuhino/status/1668575468130570241), which the artist had previously posted:
- [This tweet's image](https://twitter.com/himuhino/status/1637498607241957376) and [this one](https://twitter.com/himuhino/status/1642978925356937216) were detected as exact duplicates, but not similar images, when a reference folder is set.
- [This tweet's second image](https://twitter.com/himuhino/status/1632289429611229186) was not an exact duplicate of the other tweet, and was not detected as similar when a reference folder was set
- [This image](https://twitter.com/himuhino/status/1635145647342116867) _was_ detected as similar even with a reference folder set.
- All four were detected as similar when no reference folder was set.

[This](https://twitter.com/mitarai_4shi/status/1668557811880701952) [pair](https://www.pixiv.net/en/artworks/108982287) and [this](https://www.pixiv.net/en/artworks/108379487) [pair](https://twitter.com/yurerugin/status/1668612573892345858) of visually identical but non-exact duplicates were also only detected when the reference folder was not set.

**Comment by blazeoflate** [comment]
Yeah i'm also seeing multiple images that v5.1.0 is catching as duplicates but 6.0.0 is completely missing on the exact same setting so i've been sticking with the older version.

**Comment by anonymous721** [comment]
Both 5.1.0 and 6.0.0 miss some duplicates, but many that they miss are different from one another so running the same scan with both versions catches the widest set. 6.0.0 seems the better of the two but I also keep 5.1.0 around and when I suspect I'm missing a lot I run both versions.

More importantly though, in 6.0.0 if you set the Similarity slider at 0 it will not detect exact duplicates, and many other almost-identical images. For other values, it appears to work as expected (all duplicates detected at a threshold of 8 will also be detected at 16, for example). But adjusting it from 0 to 1 will significantly reduce how much it detects, and I've generally been running two scans, one at 0 similarity threshold and another at 40. Many of the duplicates 6.0.0 appears to fail to detect but that 5.1.0 catches are actually also findable with 6.0.0, but only with the slider at 0.

**Comment by qarmin** [comment]
Settings items as referenced items are  now checked differently, so stealing results(by other, more similar, non-referenced items) should no longer happens
